src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java

```diff
@@ -132,4 +151,50 @@ private QueryOuterClass.Query buildSelectQuery(DBFilterBase.IDFilter additionalI
         .limit(limit)
         .build();
   }
+
+  private QueryOuterClass.Query buildSortedSelectQuery(DBFilterBase.IDFilter additionalIdFilter) {
+    List<BuiltCondition> conditions = new ArrayList<>(filters.size());
+    for (DBFilterBase filter : filters) {
+      if (additionalIdFilter == null
+          || (additionalIdFilter != null && !(filter instanceof DBFilterBase.IDFilter)))
+        conditions.add(filter.get());
+    }
+    if (additionalIdFilter != null) {
+      conditions.add(additionalIdFilter.get());
+    }
+    String[] columns = sortedDataColumns;
+
+    if (orderBy() != null) {
+      List<String> sortColumns = Lists.newArrayList(columns);
+      orderBy().forEach(order -> sortColumns.addAll(order.getOrderingColumn()));
```

Comment:
addAll on a single column? or is `getOrderingColumn` returning a list? if so you would have duplicated values possible here